apps/docs/pages/guides/ai/vector-indexes/hnsw-indexes.mdx

```diff
@@ -0,0 +1,101 @@
+
+<div>
+  <img
+    alt="visual of an example skip list (light)"
+    className="dark:hidden"
+    src="/docs/img/ai/vector-indexes/hnsw-indexes/skip-list--light.png"
+  />
+  <img
+    alt="visual of an example skip list (dark)"
+    className="hidden dark:block"
+    src="/docs/img/ai/vector-indexes/hnsw-indexes/skip-list--dark.png"
+  />
+</div>
+
+When searching for an element, the algorithm begins at the top layer and traverses its linked list horizontally. If the target element is found, the algorithm stops and returns it. Otherwise if the next element in the list is greater than the target (or NIL), the algorithm drops down to the next layer below. Since each layer below is less sparse than the layer above (with the bottom layer connecting all elements), the target will eventually be found. Skip lists offer O(log n) average complexity for both search and insertion/deletion.
+
+### Navigable Small World
+
+A navigable small world (NSW) is a special type of proximity graph that also includes long-range connections between nodes. These long-range connections support the “small world” property of the graph, meaning almost every node can be reached from any other node within a few hops. Without these additional long-range connections, many hops would be required to reach a far-away node.
+
+<img
+  alt="visual of an example navigable small world graph"
+  src="/docs/img/ai/vector-indexes/hnsw-indexes/nsw.png"
+/>
+
+The “navigable” part of NSW specifically refers to the ability to logarithmically scale the greedy search algorithm on the graph, an algorithm that attempts to make only the locally optimal choice at each hop. Without this property, the graph may still be considered a small world with short paths between far-away nodes, but the greedy algorithm tends to miss them. Greedy search is ideal for NSW because it is quick to navigate and has low computational costs.
+
+### **Hierarchical +** Navigable Small World
+
+HNSW combines these two concepts. From the hierarchical perspective, the bottom layer consists of a NSW made up of short links between nodes. Each layer above “skips” elements and creates longer links between nodes further away from each other.
+
+Just like skip lists, search starts at the top layer and works its way down until it finds the target element. However instead of comparing a scalar value at each layer to determine whether or not to descend to the layer below, a multi-dimensional distance measure (such as Euclidean distance) is used.
+
+## When should you create HNSW indexes?
+
+Unlike IVFFlat indexes, you are safe to build an HNSW index immediately after the table is created. HNSW indexes are based on graphs which inherently are not affected by the same limitations as IVFFlat. As new data is added to the table, the index will be filled automatically and the index structure will remain optimal.
```

Comment:
would be good to hit two additional points here:

- When you don't need 100% accuracy and are willing to trade a small amount of accuracy for a lot of throughput
- HNSW should be your default choice